### PR TITLE
Reworked rebar and rebar rail into rebar no.5 (First time contribution)

### LINKED ISSUE #43156: Steel rail ammo density is too high.
While checking steel density I found that the material density fluctuated hugely depending on the item. The rail is one of the worst having a g/cm3 of 80!

# Steps To Reproduce

1. Look at a steel rails weight and liters volume
2. 1 liter = 1000 cm3, so 1600 cm3/ 20 cm3= 80 g/cm3

# Expected behavior

Density similar too [steel](https://amesweb.info/Materials/Density_of_Steel.aspx)

# Screenshots

steel rail

# Versions and configuration
- OS: Windows
    - OS Version: 10.0 1909
- Game Version: 0.E-5001-gec1deea [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food]
]

### PR DESCRIPTION
**Pull request by TheMasterGear**
#### Summary

SUMMARY: Balance "Corrected rebar and rail into steel, and normalized density and volume"

#### Purpose of change

Fixes #43156

#### Describe the solution

Corrected rebar and rail into steel, and normalized density and volume into the density of high carbon steel. 7.85 g/cm3 to match steel rebar type [no.5](https://www.harrissupplysolutions.com/5-rebar.html)

Also changed rebar spear not to have a reach attack, much like the pointed stick because they are both short for spears, 80 cms.

#### Describe alternatives you've considered

Could also be type [no.4](https://www.harrissupplysolutions.com/4-rebar.html), [no.6,](https://www.harrissupplysolutions.com/6-rebar.html) or any other [imperial bar size](https://www.harrissupplysolutions.com/steel-rebar-sizes-stock.html). But it's most common for [no.5](https://www.harrissupplysolutions.com/5-rebar.html) rebar to be inside roadways, which is where you get rebar in CDDA.

#### Testing
Spawn rebar and looked at the crafting menu. Made sharpened rebar and rail rebar 1% and 5% lighter for being sharpened to points and one bar of rebar makes 5 rails.

### CODE REVIEW

**Comment by TheMasterGear** [comment]
Gosh, I totally butchered getting the file up here cleanly. Sorry! Is there a way to run the game without changing the files by compiling it?

**Comment by anothersimulacrum** [comment]
I'm not sure what you're asking.

**Comment by TheMasterGear** [comment]
> 
> 
> I'm not sure what you're asking.

I used Microsoft Visual Studio to compile the GitHub version of the game to test the patch, but it changes the files slightly so I had to use a different copy of the game to test which got confusing.
changes

**Comment by anothersimulacrum** [comment]
Ah, that's weird. Well, for working on this you can just not add them (untick the checkmark there), but you should probably make an issue or seek help outside of this PR about that.

**Comment by TheMasterGear** [comment]
> 
> 
> Ah, that's weird. Well, for working on this you can just not add them (untick the checkmark there), but you should probably make an issue or seek help outside of this PR about that.

Yeah, it might be that my CMake has an error. It works also found that rail's in the JSON are ten of them when I did that calculation for only one rail so they're basic weightless now. This needs to be fixed.

File: data/json/items/ammo/metal_rail.json [hunk 1]
```diff
@@ -4,16 +4,17 @@
     "type": "AMMO",
     "name": { "str": "rebar rail" },
     "description": "A short piece of ferromagnetic rebar which has been straightened and sharpened on one end.  Could be used as a projectile by a sufficiently powerful electromagnetic weapon.",
-    "weight": "160 g",
-    "volume": "25ml",
+    "weight": "236 g",
```
Comment (anothersimulacrum) [inline data/json/items/ammo/metal_rail.json:7]: Just multiply it by the count
```suggestion
    "weight": "2360 g",
```

Comment (TheMasterGear) [reply data/json/items/ammo/metal_rail.json:7]: > 
> 
> Just multiply it by the count

 My mistake, when testing it seems that it's the volume that needs to change by a factor of 10, not the weight. The weight changes to match after this.
volume

**Comment by TheMasterGear** [comment]
It's still failing. I'm not sure what's left to fix?

**Comment by anothersimulacrum** [comment]
None of the CI failures are your fault.

**Comment by TheMasterGear** [comment]
> 
> 
> None of the CI failures are your fault.

Just out of curiosity, what when wrong then?

**Comment by anothersimulacrum** [comment]
Something changed the DPS of a melee weapon.

**Comment by TheMasterGear** [comment]
> 
> 
> None of the CI failures are your fault.

Just out of curiosity, what when wrong then?

> 
> 
> Something changed the DPS of a melee weapon.

Will I need to update my branch to fix this?

**Comment by anothersimulacrum** [comment]
Nope, CI failures that are not your fault won't cause problems for you.

**Comment by kevingranade** [comment]
Actually the test failure is related, it increased the DPS of rebar
```rebar out of range, expected: 7 got 8.40758```

**Comment by anothersimulacrum** [comment]
Whoops, missed that. I saw the header above it said club, so I thought it was that.

**Comment by kevingranade** [comment]
I read it that way the first time too.

**Comment by TheMasterGear** [comment]
Did the weight change to rebar melee need to be changed back?

**Comment by anothersimulacrum** [comment]
Probably the test needs to be changed.

**Comment by kevingranade** [comment]
Crap, I will fix this the test I promise.

File: data/json/items/generic.json [hunk 2]
```diff
@@ -709,15 +709,15 @@
...
     "category": "spare_parts",
     "price": 7500,
     "price_postapoc": 25,
     "material": [ "steel" ],
-    "weight": "908 g",
-    "volume": "1500 ml",
+    "weight": "1245 g",
+    "volume": "159 ml",
     "longest_side": "80 cm",
     "bashing": 12,
```
Comment (kevingranade) [inline data/json/items/generic.json:722]:
```suggestion
    "bashing": 10,
```

**Pull request merged**